### ISSUE TITLE
Domestic advisor: it seems that the native village tab is not showing the correct mission owner sometimes...

### ISSUE THREAD

**Issue by abmpicoli** [We-the-People-civ4col-mod/Mod#164]
Hi, everyone. Using 2.7.1 Commit efdb788e9f6dc92e3d536bc06635c62227b13bde

Found that the domestic advisor native village screen is not showing the correct mission owner. Notice that the mission owner at the map is green (portuguese? Can't say...), while the domestic advisor show it as red (danish? english? )
![image](https://user-images.githubusercontent.com/1733170/46574058-335e2f80-c974-11e8-8297-76633931970f.png)
The zip with the save (mod as "We-the-People")... 
[Domestic Advisor native village issue.zip](https://github.com/We-the-People-civ4col-mod/Mod/files/2453242/Domestic.Advisor.native.village.issue.zip)

**Comment by Nightinggale** [comment]
Looks like identical code is used to draw both icons, meaning this points towards big and small GameFont files have gone out of sync. First step for this is to implement some GameFont debug tool to allow us to see what icons the game sees at various indexes.

I wrote such a tool for M:C where an extra button shows up in the Domestic Advisor in debug mode. It then displays a list made exclusively of GameFont listings of both big and small font. It looks like I have to implement it here too or we will never really figure out why the same call (index) results in different graphics.

**Comment by Nightinggale** [comment]
I added the debug menu and it doesn't really reveal why they are different. However it looks like a lot of GameFonts are incorrectly assigned in general, not just missions.

Next up is adding a help text for mouse over to tell which player it assumes own the mission. That would be a nice addition for gameplay as well as bug hunting.

**Comment by Nightinggale** [comment]
The mystery deepens. getMissionaryChar() for Portugal returns 8641. For the city billboard, this value is set to a 16 bit character and the correct icon shows up. However to get the very same icon in python, we need to use the ID 8652. In general all missionary chars have an offset of 11, but only in python. Somebody have been trying to work around it because most civilizations have their missionary char repeated meaning they appear the same ingame despite not actually displaying the same.

In other words if we just add 11 whenever the Domestic Advisor calls getMissionaryChar(), it will work and the game will appear bugfree to the player. However I consider that to be a last resort solution. It's just hiding the real bug rather than fixing it and a hidden bug can show up again later and hurt us big time later.

What started out as a quick one line fix in familiar code has turned into a giant adventure into how the exe handles GameFont characters. The actual bug is **_NOT_** in the Domestic Advisor.

**Comment by Nightinggale** [comment]
This was one nasty bug caused by city billboards reading GameFont in a different way than the rest of the game. The result was that the icons on billboards showed the wrong GameFont for missions. Somebody had fixed this by moving the GameFont icons to the addresses actually used by the billboards, hence breaking usage of mission chars everywhere else.

I fixed the offset issue, moved icons to the correct addresses and generally cleaned up the GameFont files as they had lots of unused mission chars, like Sweden had 15 or something. Obviously this had caused a problem at some point and somebody had fixed it by spraying out mission chars all over the place to see when they showed up and didn't clean up afterwards, leaving an unreadable mess.

The good thing about this is that it forced me to write debug tools for GameFont, meaning now we can get the game to display how it view the GameFont addresses.

![python](https://user-images.githubusercontent.com/4641647/46616835-aeae1580-cb1b-11e8-9986-d9c32f7e0597.jpg)

Since city billboards apparently treat GameFont differently, I had to write debug tools for those too:
![billboard](https://user-images.githubusercontent.com/4641647/46616871-c4233f80-cb1b-11e8-94c0-f983094d30bf.jpg)

In short we should never have major problems with GameFont again.

More info on GameFont [on the wiki](https://github.com/We-the-People-civ4col-mod/Mod/wiki/GameFont).
